libres/lib/enkf/enkf_util.cpp

```diff
@@ -38,6 +38,12 @@ class generator {
     value_type operator()() { return rng_forward(rng); }
 };
 
+double enkf_util_rand_uniform(double a, double b, rng_type *rng) {
```

Comment:
Nothing wrong with this function as such - as well known I am already guilty of an excessive amount of utility functions. However I would suggest not adding this particular function because:

The Kalman Filter is obsessed with normally distributed variables, all formal results only apply with normally distributed variables (yes - this is abused left, right and center), and a common reason for poor results of various kinds is that the distributions considered are too far from normal[1]. So given that normal distribution is the "only" mathematically correct distribution to use in ERT I think it wise not to tempt following developers into making explicitly non normal code.


[1]: The main benefit/advantage of the iterated approaches is that when making small steps (i.e. iterations), the approximation of normally distributed variables is relatively better.

Reply:
I will remove it, thanks.

Is this valid for distributions used in `GEN_KW` as well?
I mean, is it not a good idea to define, say, uniformly distributed priors for parameters?

Reply:
> Is this valid for distributions used in GEN_KW as well? 

That is a good question. In a way I would like to answer yes to that question - but of course no one would appreciate a software which could only work with normally distributed values. The way this is "solved" for the GEN_KW parameters is that there is an internal underlying variable which is initially distributed as `N(0,1)` and then a transformation which transforms a variable `x ~ N(0,1)` to e.g. `U(a,b)`. The update algorithm works on the internal value, but when values are exported for the forward model they go through a transformation function.

The prior realisations internal value constitute a finite representation of the `N(0,1)` distribution, and when the update process has completed the same realisations constitute a representation of a normal distribution `N(u,w^2)` where the mean has been shifted to 0 -> u and standard deviation is `w < 1`. Now - a very important (and unfortunate) point is that the transformation which maps from e.g. `N(0,1)` to U(a,b)` only works exactly for `N(0,1)`- when the internal variables have changed mean and std dev the transformation will *not* map to `U(c,d)` with `c != a` and `d != b` - but rather a  different mathematical function.

Reply:
I need to meditate on your last comment 😄 
Thanks!